src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -148,33 +148,33 @@ private DateTime(ulong dateData)
 
         public DateTime(long ticks, DateTimeKind kind)
         {
-            if (ticks < MinTicks || ticks > MaxTicks)
-            {
-                throw new ArgumentOutOfRangeException(nameof(ticks), SR.ArgumentOutOfRange_DateTimeBadTicks);
-            }
-            if (kind < DateTimeKind.Unspecified || kind > DateTimeKind.Local)
-            {
-                throw new ArgumentException(SR.Argument_InvalidDateTimeKind, nameof(kind));
-            }
-            _dateData = ((ulong)ticks | ((ulong)kind << KindShift));
+            if ((ulong)ticks > MaxTicks) ThrowTicksOutOfRange();
+            if ((uint)kind > (uint)DateTimeKind.Local) ThrowInvalidKind();
+            _dateData = (ulong)ticks | ((ulong)(uint)kind << KindShift);
         }
 
         internal DateTime(long ticks, DateTimeKind kind, bool isAmbiguousDst)
         {
-            if (ticks < MinTicks || ticks > MaxTicks)
-            {
-                throw new ArgumentOutOfRangeException(nameof(ticks), SR.ArgumentOutOfRange_DateTimeBadTicks);
-            }
+            if ((ulong)ticks > MaxTicks) ThrowTicksOutOfRange();
             Debug.Assert(kind == DateTimeKind.Local, "Internal Constructor is for local times only");
             _dateData = ((ulong)ticks | (isAmbiguousDst ? KindLocalAmbiguousDst : KindLocal));
         }
 
+        private static void ThrowTicksOutOfRange() => throw new ArgumentOutOfRangeException("ticks", SR.ArgumentOutOfRange_DateTimeBadTicks);
+        private static void ThrowInvalidKind() => throw new ArgumentException(SR.Argument_InvalidDateTimeKind, "kind");
+        private static void ThrowMillisecondOutOfRange() => throw new ArgumentOutOfRangeException("millisecond", SR.Format(SR.ArgumentOutOfRange_Range, 0, MillisPerSecond - 1));
+
+        private static void ThrowDateArithmetic(int param)
```

Comment:
>private [](start = 8, length = 7)

nit: why didn't use same syntax of => here?

Reply:
It felt a bit long for a one liner and I saw that mostly such methods are written in the long form. I'm generally in favor of the arrow syntax (with the arrow starting on the next line for longer lines).